warehouse/forklift/legacy.py

```diff
@@ -569,7 +572,26 @@ def _is_valid_dist_file(filename, filetype):
                 }:
                     return False
 
-    if filename.endswith(".exe"):
+    tar_fn_match = _tar_filenames_re.search(filename)
+    if tar_fn_match:
+        # Ensure that this is a valid tar file, and that it contains PKG-INFO.
+        z_type = tar_fn_match.group("z_type") or ""
+        try:
+            with tarfile.open(filename, f"r:{z_type}") as tar:
+                # This decompresses the entire stream until we find the tar
+                # file member we want.  Easy to DoS attack our CPU. :(
+                bad_tar = True
+                member = tar.next()
+                while member:
+                    parts = os.path.split(member.name)
+                    if len(parts) == 2 and parts[1] == "PKG-INFO":
+                        bad_tar = False
```

Comment:
Your code comment above suggests you meant to break here. Is there a reason not to?

Reply:
I see your "review" comment now which was hidden / collapsed. I would suggest updating your code comment with an explanation of why you're not breaking (basically what is in your review comments).

Reply:
hah yes, thanks for noticing.  that's what i get for using github's inline editor to quickly remove one line.  updated! :)

Reply:
I like how you also updated `:(` to `:/`. Is that an upgrade or downgrade in the severity level? :P

Reply:
Who knows? Denial of service is always easy. The only winning move is not to serve. ;)